components/instructions/programs/dual.tsx

```diff
@@ -0,0 +1,167 @@
+import { Connection } from '@solana/web3.js'
+import {
+  StakingOptions,
+  STAKING_OPTIONS_PK,
+} from '@dual-finance/staking-options'
+import { BN, BorshInstructionCoder } from '@coral-xyz/anchor'
+import { AccountMetaData } from '@solana/spl-governance'
+import { tryGetMint } from '@utils/tokens'
+import { getMintDecimalAmountFromNatural } from '@tools/sdk/units'
+
+interface configV2Instruction {
+  lotSize: BN
+  numTokens: BN
+  optionExpiration: BN
+  soName: string
+  subscriptionPeriodEnd: BN
+}
+
+interface initStrikeInstruction {
+  strike: BN
+}
+
+interface nameTokenInstruction {
+  strike: BN
+}
+
+interface issueInstruction {
+  amount: BN
+  strike: BN
+}
+
+const INSTRUCTIONS = {
+  225: {
+    name: 'Staking Option ConfigV2',
+    accounts: [
+      { name: 'authority' },
+      { name: 'soAuthority' },
+      { name: 'issueAuthority' },
+      { name: 'state' },
+      { name: 'baseVault' },
+      { name: 'baseAccount' },
+      { name: 'quoteAccount' },
+      { name: 'baseMint' },
+      { name: 'quoteMint' },
+      { name: 'tokenProgram' },
+      { name: 'systemProgram' },
+      { name: 'rent' },
+    ],
+    getDataUI: async (
+      connection: Connection,
+      data: Uint8Array,
+      accounts: AccountMetaData[]
+    ) => {
+      const soHelper = new StakingOptions(connection.rpcEndpoint)
+
+      const decodedInstructionData = new BorshInstructionCoder(
+        soHelper.getIdl()
+      ).decode(Buffer.from(data))?.data as configV2Instruction
+
+      const baseMint = await tryGetMint(connection, accounts[7].pubkey)
+
+      const rawAmount = decodedInstructionData.numTokens
+      const tokenAmount = baseMint
+        ? getMintDecimalAmountFromNatural(baseMint.account, rawAmount)
+        : rawAmount
+
+      return (
+        <div className="space-y-3">
+          <div>
+            Expiration:{' '}
+            {new Date(
+              decodedInstructionData.optionExpiration.toNumber() * 1000
+            ).toDateString()}
+          </div>
+          <div>
+            Subscription Period End:{' '}
+            {new Date(
+              decodedInstructionData.subscriptionPeriodEnd.toNumber() * 1000
+            ).toDateString()}
+          </div>
+          <div>Num Tokens: {tokenAmount.toNumber()}</div>
```

Comment:
Will this be in tokens or atoms or options contracts? Rename Num Option Tokens?

Reply:
The calculation above makes this tokens